be/src/exec/stream/aggregate/stream_aggregator.h

```diff
@@ -0,0 +1,151 @@
+                           std::vector<vectorized::ChunkPtr> detail_chunks);
+
+    // Only create selected agg funcs by agg_func_ids
+    vectorized::Columns _create_needed_agg_result_columns(size_t num_rows,
+                                                          const std::vector<int32_t>& agg_func_ids,
+                                                          bool use_intermediate);
+
+    vectorized::ChunkPtr _build_output_chunk_with_ops(const vectorized::Columns& group_by_columns,
+                                                      const vectorized::Columns& agg_result_columns,
+                                                      const UInt8ColumnPtr& ops,
+                                                      bool is_intermediate_result);
+
+    Status _output_intermediate_changes(int32_t chunk_size,
+                                        const vectorized::Columns& group_by_columns,
+                                        vectorized::ChunkPtr* intermediate_chunk);
+
+    Status _output_result_changes(int32_t chunk_size,
+                                  const vectorized::Columns& group_by_columns,
+                                  vectorized::ChunkPtr* intermediate_chunk);
+    Status _output_result_post_retract_changes(size_t chunk_size,
+                                               const vectorized::Columns& group_by_columns,
+                                               const vectorized::Buffer<uint8_t>& prev_existence,
+                                               const ChunkPtr& prev_result_chunk,
+                                               ChunkPtr* result_chunk);
+    Status _output_result_prev_retract_changes(size_t chunk_size,
+                                               const vectorized::Columns& group_by_columns,
+                                               ChunkPtr* prev_result_chunk,
+                                               vectorized::Buffer<uint8_t>* prev_existence);
+    Status _output_result_changes_without_retract(size_t chunk_size,
+                                                  const vectorized::Columns& group_by_columns,
+                                                  ChunkPtr* result_chunk);
+
+private:
+    // Store buffers which can be reused in the incremental compute.
+    std::unique_ptr<MemPool> _mem_pool;
+
+    // Store group by keys to agg state map.
+    std::unique_ptr<StateTable> _result_state_table;
+    std::unique_ptr<StateTable> _intermediate_state_table;
+    // TODO: support merge into one detail table later.
+    std::vector<std::unique_ptr<StateTable>> _detail_state_tables;
+    std::vector<SlotDescriptor*> _output_slots_without_op;
+    int32_t _count_agg_idx{0};
+
+    // store all agg states
+    std::vector<std::unique_ptr<AggStateData>> _agg_func_states;
+    // output intermediate columns.
+    std::vector<int32_t> _intermediate_agg_func_ids;
+
+    std::unique_ptr<IntermediateAggGroupState> _result_agg_group;
```

Comment:
Because result/intermediate state's process policy are the same, just the state table is different, so `_result_agg_group ` just reuse `IntermediateAggGroupState `